frontend/src/components/search/SearchBar.tsx

```diff
@@ -1,179 +1,265 @@
+    modes: {
+      field: "mode",
+      description: "car, 2W, 3W, multi-modal etc.",
+      options: data?.facets?.modes?.items,
+      isMultiple: true,
+    },
+    services: {
+      field: "service",
+      description: "passenger or freight",
+      options: data?.facets?.services?.items,
+      isMultiple: true,
+    },
+  };
+
+  /* handle click outside the search container to close command list */
   useEffect(() => {
     const handleClickOutside = (event: MouseEvent) => {
       if (
         commandRef.current &&
         !commandRef.current.contains(event.target as Node)
       ) {
-        if (!hideDatasetSuggestion) setShowCommandList(false);
+        setShowCommandList(false);
       }
     };
-
     document.addEventListener("mousedown", handleClickOutside);
     return () => {
       document.removeEventListener("mousedown", handleClickOutside);
     };
   }, []);
+  //get stored searches
+  useEffect(() => {
+    if (typeof window !== "undefined")
+      setStoredSearches(
+        JSON.parse(localStorage?.getItem("tdcRecentSearches") ?? "[]")
+      );
+  }, []);
 
-  const handleNarrowSelect = (facet: any) => {
-    if (facet) {
-      onChange([{ value: facet, key: facetName! }]);
-      queryInputRef.current?.focus();
-      // setFocus("query");
-    }
+  useEffect(() => {
+    const setQueryValue = setTimeout(() => {
+      setQuery(inputValue);
+    }, 400);
+
+    return () => clearTimeout(setQueryValue);
+  }, [inputValue]);
+
+  const handleFacetNameChange = (facetName: string) => {
+    if (facetName) setFacetName(facetName);
+  };
+
+  const handleFacetValueChange = (item: FacetValueProps) => {
+    setFacetValue(item);
+    inputRef?.current?.focus();
   };
 
   const handleTyping = (value: string) => {
-    setInternalQuery(value);
+    setInputValue(value);
     setIsTyping(value.length > 0);
   };
 
-  useEffect(() => {
-    if (!internalQuery) setInternalQuery(query);
-  }, [query]);
-
   const handleCancelSearch = () => {
-    setInternalQuery("");
-    const changes: any = [];
-    if (facetName) changes.push({ value: undefined, key: facetName });
-    changes.push({ value: undefined, key: "query" });
-    onChange(changes);
-    setTimeout(() => {
-      queryInputRef.current?.focus();
-      // setFocus("query");
-    }, 150);
+    setFacetName("");
+    setFacetValue({
+      display_name: "",
+      name: "",
+    });
+    setInputValue("");
+    setQuery("");
+    setIsTyping(false);
+  };
+
+  const handleSubmit = () => {
+    const value = facets[facetName]?.isMultiple
+      ? [facetValue.name]
+      : facetValue.name;
+
+    const queryObject = {
+      ...(facetValue.name ? { [facetName]: value } : {}),
+      query: inputValue,
+    };
+    if (inputValue) storeRecentSearch(queryObject);
+
+    setShowCommandList(false);
+
+    router.push({
+      pathname: "/search",
+      query: queryObject,
+    });
+
+    return false;
+  };
+
+  const storeRecentSearch = (search: RecentSearchProps) => {
```

Comment:
The logic implemented here needs to be changed to work as the mockup:

![image](https://github.com/user-attachments/assets/db041af9-2ac0-4e58-b00e-0de9cabe635c)

Currently the search stores just the texts searched but ignoring the facets used to search:

![image](https://github.com/user-attachments/assets/1fcd66dd-8c0e-42be-a250-431a7dd0af8b)

Reply:
fixed